### PR TITLE
add plain provisioner http server test

### LINKED ISSUE #427: Add e2e tests for the content http server of the plain provisioner
Add e2e tests that exercise the content http server of the plain provisioner (including authorization via kube-rbac-proxy).

Follow-up work from #311

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

Closes #427

### CODE REVIEW

File: test/e2e/plain_provisioner_test.go [hunk 1]
```diff
@@ -177,6 +180,124 @@ var _ = Describe("plain provisioner bundle", func() {
...
+					},
+				}
+				err = c.Create(ctx, &job)
+				Expect(err).To(BeNil())
+				Eventually(func() (bool, error) {
+					err = c.Get(ctx, types.NamespacedName{Name: "rukpak-svr-job", Namespace: defaultSystemNamespace}, &job)
+					if err != nil {
+						return false, err
+					}
+					return job.Status.CompletionTime != nil, err
```
Comment (exdx) [inline test/e2e/plain_provisioner_test.go:264]: should we check the job completion phase is success as well here?

File: test/e2e/plain_provisioner_test.go [hunk 2]
```diff
@@ -177,6 +180,124 @@ var _ = Describe("plain provisioner bundle", func() {
...
+					// Get logs of the Pod
+					pod = pods.Items[0]
+					logReader, err := kubeClient.CoreV1().Pods(defaultSystemNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{}).Stream(context.Background())
+					if err != nil {
+						return false, err
+					}
+					buf := new(bytes.Buffer)
+					_, err = buf.ReadFrom(logReader)
+					Expect(err).To(BeNil())
+					return strings.Contains(buf.String(), "manifests/crds.yaml"), nil
```
Comment (exdx) [inline test/e2e/plain_provisioner_test.go:289]: can we check against all the files we expect in the bundle here? or is this the only one

File: test/e2e/plain_provisioner_test.go [hunk 3]
```diff
@@ -177,6 +180,124 @@ var _ = Describe("plain provisioner bundle", func() {
...
+				}).Should(BeTrue())
+			})
+			err := c.Delete(ctx, &job)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &pod)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &crb)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &sa)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
```
Comment (exdx) [inline test/e2e/plain_provisioner_test.go:299]: I would suggest wrapping these delete statements in an Eventually() and check for `client.IgnoreNotFound(err) == true` to ensure everything is removed before ending the test.

File: test/e2e/plain_provisioner_test.go (hunk 2 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go:289]: This bundle actually has only one file in it.  The http server provides tgz file so if this check sees a file in the tgz file, the http server is working.  We can move this test under another test that has multiple files in the bundle.

File: test/e2e/plain_provisioner_test.go [hunk 4]
```diff
@@ -494,6 +497,130 @@ var _ = Describe("plain provisioner bundle", func() {
...
+					err = c.Create(ctx, &job)
+					Expect(err).To(BeNil())
+					Eventually(func() (bool, error) {
+						err = c.Get(ctx, types.NamespacedName{Name: "rukpak-svr-job", Namespace: defaultSystemNamespace}, &job)
+						if err != nil {
+							return false, err
+						}
+						return job.Status.CompletionTime != nil && job.Status.Succeeded == 1, err
+					}).Should(BeTrue())
+				})
```
Comment (timflannagan) [inline test/e2e/plain_provisioner_test.go:584]: Move this to a BeforeEach clause?

**Review by timflannagan — COMMENTED** [review]
This generally looks reasonable to me, but I think we'll need to refactor the test setup to use BeforeEach/AfterEach clauses to properly clean up the test artifacts.

**Review by tylerslaton — APPROVED** [review]
Super small nit, but otherwise looks good!

File: test/e2e/plain_provisioner_test.go [hunk 5]
```diff
@@ -494,6 +497,130 @@ var _ = Describe("plain provisioner bundle", func() {
...
+
+						Subjects: []rbacv1.Subject{{Kind: "ServiceAccount", Name: "rukpak-svr-sa", Namespace: defaultSystemNamespace}},
+						RoleRef:  rbacv1.RoleRef{APIGroup: "rbac.authorization.k8s.io", Kind: "ClusterRole", Name: "bundle-reader"},
+					}
+
+					err = c.Create(ctx, &crb)
+					Expect(err).To(BeNil())
+					url := bundle.Status.ContentURL
+
+					// Create a Job that reada from the URL and outputs contents in the pod log
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go:551]: I fix it.  Thanks!

File: test/e2e/plain_provisioner_test.go (hunk 4 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go:584]: Thanks for the idea.  It can clean up the resources at test failure now.

**Pull request merged**